editoast/src/views/search/config.rs

```diff
@@ -25,9 +25,33 @@ pub struct SearchResult {
     pub columns: HashMap<String, String>,
 }
 
+#[derive(Debug, Deserialize, Clone, PartialEq)]
+#[serde(untagged)]
+pub enum TypeDescriptor {
+    Atomic(AstType),
+    Compound(CompoundTypeDescriptor),
+}
+
+#[derive(Debug, Deserialize, Clone, PartialEq)]
+#[serde(tag = "type", rename_all = "lowercase")]
+pub enum CompoundTypeDescriptor {
+    Array { item: Box<TypeDescriptor> },
+}
```

Comment:
We can make it more consistent by requiring the `type` tag for all columns. Or untag the compound types. No matter which solution is chosen we will end with only one struct.